misc/python/materialize/data_ingest/data_type.py

```diff
@@ -0,0 +1,187 @@
+        raise NotImplementedError
+
+
+class NullType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        return None
+
+
+class BooleanType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        return random.choice((True, False))
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        return "boolean"
+
+
+class IntType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.randint(-127, 128)
+        elif record_size == RecordSize.SMALL:
+            return random.randint(-32768, 32767)
+        elif record_size == RecordSize.MEDIUM:
+            return random.randint(-2147483648, 2147483647)
+        elif record_size == RecordSize.LARGE:
+            return random.randint(-9223372036854775808, 9223372036854775807)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        return "int"
+
+
+class LongType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.randint(-127, 128)
+        elif record_size == RecordSize.SMALL:
+            return random.randint(-32768, 32767)
+        elif record_size == RecordSize.MEDIUM:
+            return random.randint(-2147483648, 2147483647)
+        elif record_size == RecordSize.LARGE:
+            return random.randint(-9223372036854775808, 9223372036854775807)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "long"
+        else:
+            return "bigint"
+
+
+class FloatType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
+        if record_size == RecordSize.TINY:
+            return random.random()
+        elif record_size == RecordSize.SMALL:
+            return random.uniform(-100, 100)
+        elif record_size == RecordSize.MEDIUM:
+            return random.uniform(-1_000_000, 1_000_000)
+        elif record_size == RecordSize.LARGE:
+            return random.uniform(-1_000_000_000, 1_000_000_000_00)
+        else:
+            raise ValueError(f"Unexpected record size {record_size}")
+
+    @staticmethod
+    def num_value(num: int) -> Any:
+        return num
+
+    @staticmethod
+    def name(backend: Backend) -> str:
+        if backend == Backend.AVRO:
+            return "float"
+        else:
+            return "float4"
+
+
+class DoubleType(DataType):
+    @staticmethod
+    def random_value(record_size: RecordSize) -> Any:
```

Comment:
Also identical to `FloatType`?

Reply:
True, made it derive from FloatType. Initially I was planning for more difference, but want to get this in before it grows too large.